sdk/scheduler/src/main/java/com/mesosphere/sdk/config/DefaultConfigurationUpdater.java

```diff
@@ -83,6 +78,31 @@ public UpdateResult updateConfiguration(ServiceSpec candidateConfig) throws Conf
                     String.format("Unable to serialize new config to JSON for logging: %s", e.getMessage())));
         }
 
+        if (targetConfig == null) {
+            LOGGER.info("Skipping config diff: There is no old config target to diff against");
+        } else if (candidateConfigJson == null) {
+            LOGGER.info("Skipping config diff: New target couldn't be represented as JSON");
```

Comment:
When could this happen?  Should we fail hard here, or is there YAML in the world which can validly not be expressed as JSON?